jekyll/_cci2/caching-strategy.md

```diff
@@ -16,16 +16,66 @@ Caching is one of the most effective ways to make jobs faster on CircleCI. By re
 {:toc}
 
 ## Cache optimization
-When setting up caches for your projects, the goal would be a 10x - 20x ROI (return on investment). The following tips can help you achieve this.
+{: #cache-optimization }
+
+When setting up caches for your projects, the goal would be a 10x - 20x ROI (return on investment). This means you should be aiming for a situation where the amount of cache restored should be 10x - 20x the cache saved. The following tips can help you achieve this.
+
+### Avoid strict cache keys
+{: #avoid-strict-cache-keys }
+
+Using cache keys that are too strict can mean that you will only get a minimal number of cache hits for a workflow. For example, if you used the key `CIRCLE_SHA1` (SHA of the last commit of the current pipeline) this would only get matched once for a workflow. Consider using cache keys that are less strict to ensure more cache hits.
 
-### Use cache keys that are less strict
 ### Avoid unnecessary workflow reruns
+{: #avoid-unnecessary-workflow-reruns }
+
+If your project has "flaky tests" workflows might be rerun unnecessarily. This will both use up your credits and increase your storage usage. To avoid this situation, address flaky tests. For help with identifying them, see [Test Insights]({{ site.baseurl }}/2.0/insights-tests/#flaky-tests)). You can also consider configuring your projects to only rerun failed jobs rather than entire workflows. To achieve this you can use the `when` step. For further information see the [Configuration Reference]({{ site.baseurl }}/2.0/configuration-reference/#the-when-attribute).
```

Comment:
```suggestion
If your project has "flaky tests," workflows might be rerun unnecessarily. This will both use up your credits and increase your storage usage. To avoid this situation, address flaky tests. For help with identifying them, see [Test Insights]({{ site.baseurl }}/2.0/insights-tests/#flaky-tests)). You can also consider configuring your projects to only rerun failed jobs rather than entire workflows. To achieve this you can use the `when` step. For further information see the [Configuration Reference]({{ site.baseurl }}/2.0/configuration-reference/#the-when-attribute).
```